werror_test.go

```diff
@@ -560,3 +560,25 @@ func TestWrapPullsOutParamsFromContext(t *testing.T) {
 	assert.Equal(t, map[string]interface{}{"safeKey": "safeValue", "anotherSafeKey": "anotherSafeValue", "aThirdSafeKey": "aThirdSafeValue"}, safeFromError)
 	assert.Equal(t, map[string]interface{}{"unsafeKey": "unsafeValue", "anotherUnsafeKey": "anotherUnsafeValue", "aThirdUnsafeKey": "aThirdUnsafeValue"}, unSafeFromError)
 }
+
+func TestSafeParams(t *testing.T) {
+	err := werror.Error("first", werror.SafeParam("key", "value"))
+	err2 := werror.Wrap(err, "second", werror.SafeParam("key2", "value2"))
+	err3 := werror.Wrap(err2, "second", werror.SafeParam("key3", "value3"))
+	assert.Equal(t, map[string]interface{}{"key": "value", "key2": "value2"}, err2.(werror.Werror).SafeParams())
+	assert.Equal(t, map[string]interface{}{"key": "value", "key2": "value2", "key3": "value3"}, err3.(werror.Werror).SafeParams())
+
+	err4 := werror.Wrap(err3, "second", werror.SafeParam("key", "value4"))
```

Comment:
nit could you add a short comment above these saying that they're checking for the deepest param or that this param is a noop? Took me a sec to grok 😅